test(dla): document dla limitations. (#68)
flatland-association/flatland-baselines: flatland_baselines/deadlock_avoidance_heuristic/README.md

#### Algorithm

> [!NOTE]  
> Do not enter/move if there is no free space between my path and any oncoming train's path. (A train is oncoming if it is on my path and running towards me).

#### Guarantee

> As long as trains “see” each other, they can avoid each other if there were no other trains blocking them.

Author: ```suggestion
As long as (i) trains “see” each other and (ii) there are no further trains blocking them, there is at least one non-overlapping segment between two trains where they can evade each other.```

> #### Limits

Author: Add limit: no support for loopy paths?

> ##### Known weaknesses: jamming

Author: ```suggestion
##### Known weakness: jamming
```

conflict for either of them and issues MOVE_FORWARD for both. It is the environment's own motion check, not this
method, that then admits only one of them into the cell for that step; the other is simply stopped in place. See
`tests/test_limitations_switch.py`.

Similarly, two agents approaching each other through switches at both ends
of a single-track segment with no passing loop and no alternative route between them will permanently deadlock
at their own switch cell -- one step short of ever entering the shared segment between the switches. Since their
paths are exact mirror images of one another over that entire segment, there is no free cell for either of them
to wait on inside it, so the pairwise-deadlock-free guarantee degrades to both agents blocking each other forever
right at the entrance. This is an instance of the "Require alternative path" problem class described above, which
this method cannot resolve on its own -- it only avoids the collision, it does not free the trains. See
`tests/test_limitations_two_switches.py`.

##### Known weakness: entering at the same time

Use `entering_prevention=True` to check for agents entering at the same time leading to a deadlock.

> 

Author: ```suggestion

##### Known weakness: loopy paths

Loopy paths are not supported.

```
